manager/allocator/networkallocator/networkallocator.go

```diff
@@ -303,38 +324,81 @@ func OnInit(options *ServiceAllocationOpts) {
 // IsServiceAllocated returns if the passed service has its network resources allocated or not.
 // init bool indicates if the func is called during allocator initialization stage.
 func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, flags ...func(*ServiceAllocationOpts)) bool {
-	var options ServiceAllocationOpts
+	return !na.ServiceNeedsAllocation(s, flags...)
+}
```

Comment:
Please remove this function and change its callers to use `!na.ServiceNeedsAllocation(s, ...)` directly.